### DISCUSSION TITLE
Using variables in a Text element

### DISCUSSION THREAD

**Discussion by norricorp** [slint-ui/slint#4681] [Q&A]
I used the template to create a slint project. That has a slint file and a main.rs file. The default code has a text element that displays an incremented counter variable.
Text {  text: "Counter: \{root.counter}";  }
Obviously it works.

I thought it would be interestng to write it with a single main.rs file that has a slint! macro section. So in the style of the memory game tutorial.
The above code fails compilation because of the backslash. So escape that and it compiles. But the app displays as "Counter: \{root.counter}". No backslashes gives same without a backslash.
If I code it as 
				Text {
//					text: "Counter: \\{root.counter}";
					text: root.counter;
				}
then compiles and I get a number displayed in the app.

What am I utterly misunderstanding here?

**Comment by norricorp** [comment]
So instead I wrote it as 
text: "Counter: " + root.counter;
and that worked.

**Reply by Vadoola** [reply]
As far as I'm aware in order for the variable to be display it needs to be escaped with \{varname}.

Looking at the [AppWindow.slint](https://github.com/slint-ui/slint-rust-template/blob/main/ui/appwindow.slint) in the Slint Rust template on Github, it also includes the \:

`Text {
    text: "Counter: \{root.counter}";
}`

Are you saying the version you created your template from didn't include the \ in front of {root.counter} and still worked? As far as I understand it's needed, and the template as shown on Github has it.

**Comment by ogoffart** [comment]
The `\{...}`  syntax don't work in a .rs file in a slint! macro because that cannot be tokenized by the rust compiler.  So the workaround is to use `+` as you did.


(this is mentioned there: https://slint.dev/releases/1.4.1/docs/slint/src/language/syntax/types#strings )

**Reply by Vadoola** [reply]
Ah, I misread his question, not it makes more sense to me. I actually didn't know you couldn't use the \{...} in the macro either because I haven't used the macro's much. Good to know for the future.
